### PR TITLE
[8.0][WIP] Added Earned Value Management Report

### PR DESCRIPTION
**Pull request by sudhir-serpentcs**
@elicoidal   Added Earned Value Management Report.

### CODE REVIEW

File: business_requirement_earned_value/report/br_earned_value_report.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    remaining_hours = fields.Float('Remaining time', readonly=True)
+    total_expected_time = fields.Float('Total expected time', readonly=True)
+    project_completion = fields.Float('Project Completion', readonly=True)
+    earned_value = fields.Float('Earned Value', readonly=True)
+
+    def init(self, cr):
+        tools.drop_view_if_exists(cr,
+                                  'business_requirement_earned_value_report')
+        cr.execute("""
+            CREATE VIEW business_requirement_earned_value_report AS (
```
Comment (elicoidal) [inline business_requirement_earned_value/report/br_earned_value_report.py]: I see a lot of div by zero potential issues in the current view.

File: business_requirement_deliverable_cost/models/business.py [hunk 2]
```diff
@@ -181,7 +181,7 @@ def _compute_resource_task_total(self):
                     br.mapped('deliverable_lines').mapped(
                         'resource_ids').filtered(
                         lambda r: r.resource_type == 'task').mapped(
-                            'price_total'))
```
Comment (elicoidal) [inline business_requirement_deliverable_cost/models/business.py]: irrelevant

File: business_requirement_earned_value/report/__init__.py [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# -*- coding: utf-8 -*-
+# © 2016 Elico Corp (www.elico-corp.com).
```
Comment (elicoidal) [inline business_requirement_earned_value/report/__init__.py]: 2017 (everywhere)

**Review by elicoidal — CHANGES_REQUESTED** [review]
Some remarks.
Besides, to be tested.

**Comment by sudhir-serpentcs** [comment]
@elicoidal 
Please review code.

File: business_requirement_earned_value/report/br_earned_value_report_view.xml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+        <record id="action_earned_value_report" model="ir.actions.act_window">
+            <field name="name">Earned Value Analysis</field>
+            <field name="res_model">business.requirement.earned.value.report</field>
+            <field name="view_type">form</field>
+            <field name="view_mode">graph</field>
+            <field name="view_id" ref="view_earned_value_graph"/>
+        </record>
+
+        <menuitem action="action_earned_value_report" id="menu_action_deliverable_report"
```
Comment (elicoidal) [inline business_requirement_earned_value/report/br_earned_value_report_view.xml]: This report should only visible to the cost group

**Review by elicoidal — CHANGES_REQUESTED** [review]
Almost there!

**Comment by elicoidal** [comment]
almost there!

**Comment by elicoidal** [comment]
please check Travis and Runbot as well

File: business_requirement_earned_value/report/br_earned_value_report.py [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+    name = fields.Char('Name', readonly=True)
+    description = fields.Char('Description', readonly=True)
+    partner_id = fields.Many2one('res.partner', 'Customer',
+                                 readonly=True)
+    project_id = fields.Many2one('project.project', 'Master Project',
+                                 readonly=True)
+    res_product = fields.Many2one('product.product', 'Res Product',
+                                  readonly=True)
+    hr_timesheet_product =\
```
Comment (martintorresvm) [inline business_requirement_earned_value/report/br_earned_value_report.py]: @sudhir-serpentcs layout is not ok.
as example do as here:
hr_timesheet_product = fields.Many2one(
    'product.product',
    'HR Timesheet Product',
    ...,)

And amend for the others fields as well

File: business_requirement_earned_value/report/br_earned_value_report.py [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+            where id = aal.product_id)) as actual_cost,
+            ((pt.effective_hours * (select list_price from product_template
+            where id = aal.product_id)) - (res.qty * res.unit_price))
+            as variance,
+            (((pt.effective_hours * (select list_price from product_template
+            where id = aal.product_id)) - (res.qty * res.unit_price)) /
+            res.unit_price) as per_variances,
+            pt.remaining_hours,
+            (pt.effective_hours + pt.remaining_hours) as total_expected_time,
+            CASE WHEN (pt.effective_hours + pt.remaining_hours) > 0 THEN
```
Comment (martintorresvm) [inline business_requirement_earned_value/report/br_earned_value_report.py]: @sudhir-serpentcs if 
`pt.effective_hours = 0`
`pt.remaining_hours = 2`

condition `(pt.effective_hours + pt.remaining_hours) > 0` is true then a division 0/2 on Postgress will cause error, don't you think?

**Review by martintorresvm — CHANGES_REQUESTED** [review]
Please have a check and amend if necessary

File: business_requirement_earned_value/report/br_earned_value_report.py (hunk 6 above)
Comment (YogeshMahera-SerpentCS) [inline business_requirement_earned_value/report/br_earned_value_report.py]: @victormartinelicocorp 

no if 0/2
condition should be (pt.remaining_hours) > 0

Comment (YogeshMahera-SerpentCS) [reply business_requirement_earned_value/report/br_earned_value_report.py]: @victormartinelicocorp 

no if 0/2
condition should be (pt.remaining_hours) > 0

Comment (martintorresvm) [reply business_requirement_earned_value/report/br_earned_value_report.py]: Yes, please amend

**Comment by sudhir-serpentcs** [comment]
@elicoidal @victormartinelicocorp 
Added group and Improved code.
please review code.

File: business_requirement_earned_value/report/br_earned_value_report_view.xml [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+            parent="base.menu_project_report" groups="business_requirement_deliverable_cost.group_business_requirement_cost_control"/>
```
Comment (elicoidal) [inline business_requirement_earned_value/report/br_earned_value_report_view.xml]: Please improve formatting (one attribute one line)

File: business_requirement_earned_value/__openerp__.py [hunk 8]
```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
+# © 2017 Elico Corp (https://www.elico-corp.com).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+{
+    'name': ' Earned Value Management',
+    'category': 'Business Requirements Management',
+    'summary': 'Manage the Business Requirement Deliverables and \
```
Comment (elicoidal) [inline business_requirement_earned_value/__openerp__.py]: BI report for the based on Earned Value Management

**Review by elicoidal — CHANGES_REQUESTED** [review]
Please add a README file (based on business_requirement module) so that I can create the content

**Comment by nikul-serpentcs** [comment]
@elicoidal @victormartinelicocorp 

Added README.rst  and Improved code.

**Comment by sudhir-serpentcs** [comment]
@elicoidal @victormartinelicocorp 

Added README.rst  and Improved code.

**Review by elicoidal — APPROVED** [review]

**Comment by elicoidal** [comment]
I made some functional tests and I have several feedbacks.

1. I cannot relate the report result with the set up I did in the BR.
Here are the DL/RL (101 in total)
screenshot from 2017-04-09 19-28-43
screenshot from 2017-04-09 19-28-32

Here are the timesheets I have inputed:
screenshot from 2017-04-09 19-27-03

Here is the result:
screenshot from 2017-04-09 19-26-23

I would have expected: 
* Planned Hours: 101 instead of 404
* Product cost: 25.099 (is a calculation as we cannot get actual cost for multiple lines: cost/hours)
* Cost: 2535 instead of 10140
* Actual time: 60 instead of 240
* Product cost from TS: 30 instead 600 (could be calculation: Actual cost / Actual time)
* Actual Cost: 60x30= 1800 instead of 18000
(rest of column are obviously calculated).

2. Besides can you change the following column string:
* Planned Time in RL: Planned Time
* Product Cost in RL: Plan. Unit Cost
* Actual Time in Timesheet => Actual Time
* Product Cost from Timesheet Product => Act. Unit Cost

3. Other remarks
* If there is no timesheet, the report is empty: it should display the BR content with actual time = 0
* Name: should be BR name+description

**Review by elicoidal — CHANGES_REQUESTED** [review]
Failed functional tests

**Comment by sudhir-serpentcs** [comment]
@elicoidal @victormartinelicocorp 
please review code.

**Comment by elicoidal** [comment]
@sudhir-serpentcs 
Retested and still not correct (although seems improving: now it is double instead of quadruple...).
screenshot from 2017-04-13 09-12-14

Please check the test I have written in https://github.com/OCA/business-requirement/pull/107#issuecomment-292781110 before submitting for review.

**Comment by sudhir-serpentcs** [comment]
@elicoidal @victormartinelicocorp 

please review code

**Comment by elicoidal** [comment]
@sudhir-serpentcs it is better. Some additional feedback:
- [X] Please update the titles as in https://github.com/OCA/business-requirement/pull/107#issuecomment-292781110
- [X] Project completion should appear as 81 not 0.81 (like % variance)
- [X] Variance should be reversed (negative)
- [X] Earned Value column seems incorrect

**Comment by sudhir-serpentcs** [comment]
@elicoidal @victormartinelicocorp

please review code

**Comment by elicoidal** [comment]
For https://github.com/OCA/business-requirement/pull/107#issuecomment-294069860 all is clear!
Small details I want to adjust:
* Total expected time => Total Exp. time
* % Project Completion => % Completion
After that, good to merge :)

**Comment by sudhir-serpentcs** [comment]
@elicoidal @victormartinelicocorp

please review code

**Comment by elicoidal** [comment]
Good job! @sudhir-serpentcs

**Review by elicoidal — APPROVED** [review]

**Comment by elicoidal** [comment]
@seb-elico @victormartinelicocorp can you review?

**Comment by elicoidal** [comment]
@pedrobaeza @dreispt @seb-elico @victormartinelicocorp Let's move forward?

File: business_requirement_deliverable_cost/models/business.py (hunk 2 above)
Comment (pedrobaeza) [reply business_requirement_deliverable_cost/models/business.py]: Don't make this change in the PR.

**Review by pedrobaeza — APPROVED** [review]

**Comment by sudhir-serpentcs** [comment]
@pedrobaeza @elicoidal 

please review code

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**